packages/pkl.lua/tests/parser.pkl-expected.pcf

```diff
@@ -0,0 +1,232 @@
+examples {
+  ["empty"] {
+    new {}
+  }
+  ["null"] {
+    new {
+      foo = null
+    }
+  }
+  ["boolean"] {
+    new {
+      foo = true
+      bar = false
+    }
+  }
+  ["number"] {
+    new {
+      zero = 0
+      one = 1
+      negative = -1
+      maxInt32 = 2147483647
+      minInt32 = -2147483648
+      zerof = 0.0
+      float = 5.32
+      negfloat = -10.26
+      hex = 43981
+      hexf = 128.0
+      hexf2 = 128.5
+      hexf3 = 127.9375
+      hexf4 = 127.515625
+      hexf5 = 127.6708984375
+      hexp = 2.0
+      hexp2 = 4.0
+      hexp3 = 0.5
+      hexp4 = 257.0
+    }
+  }
+  ["string"] {
+    new {
+      s = "hello world"
+      single = "one'two"
+      double = "one\"two"
+      escapes = "\u{7}\u{8}\u{c}\n\r\t\u{b}"
+      hex = "\u{0}::\u{7f}"
+      dec = "\u{0}:::0\u{7f}"
+      u = "\u{0}《》\u{10FFFF}"
+      newline = "foo\n  bar"
+      z = "foobar"
+      long = "foo"
+      long2 = "foo"
+      long3 = "  foo"
+      long4 = "]]"
+      long5 = "]==]]====]"
+    }
+  }
+  ["errors"] {
+    // note: module.catch is replacing newlines with spaces for some reason
+    // so we can't quite assert the full formatting
+    """
+    expected value or {, found EOF
+
+    1 | foo=
+      |     ^
+    at <input>:1:5
+    """.replaceAll("\n", " ")
+    """
+    expected value or {, found EOF
+
+    2 | bar=
+      |     ^
+    at <input>:2:5
+    """.replaceAll("\n", " ")
+    """
+    invalid numeric literal: 1.2.3
+
+    1 | foo=1.2.3
+      |     ^^^^^
+    at uri:path/to/input.pkl:1:5
+    """.replaceAll("\n", " ")
+    """
+    table has both list elements and map entries
+
+    1 | foo={1, [true]=2}
+      |      ^ first list entry
+      |          ^^^^ first map entry
+    at <input>:1:6
+    """.replaceAll("\n", " ")
+    """
+    table has both list elements and map entries
+
+    1 | foo={[true]=1, 2}
+      |       ^^^^ first map entry
+      |                ^ first list entry
+    at <input>:1:7
+    """.replaceAll("\n", " ")
+    """
+    table has both list elements and map entries
+
+    2 |   [true]=1;
+      |    ^^^^ first map entry
+    3 |   2
+      |   ^ first list entry
+    at <input>:2:4
+    """.replaceAll("\n", " ")
+    """
+    table has both list elements and map entries
+
+    2 |   1;
+      |   ^ first list entry
+    3 |   [true]=2
+      |    ^^^^ first map entry
+    at <input>:2:3
+    """.replaceAll("\n", " ")
+    """
+    expected identifier or ;, found token `[`
+
+    1 | [1]=1
+      | ^
+    at <input>:1:1
+    """.replaceAll("\n", " ")
+    """
+    _ENV cannot be assigned to directly
+
+    1 | _ENV=1
+      |     ^
+    at <input>:1:5
+    """.replaceAll("\n", " ")
```

Comment:
Very cool! Great job with the error reporting here; this looks great.

Per my other comment, we can't have method calls here; this is not valid PCF.

I think let's make these facts, and do a string equality in your tests.

For example:

```groovy
facts {
  ["errors"] {
    module.catch(() -> parser.parse("_ENV=1")) == """
      _ENV cannot be assigned to directly

      1 | _ENV=1
        |     ^
      at <input>:1:5
      """.replaceAll("\n", " ")
  }
}
```

Reply:
Done